diff --git a/scripts/package/mkdebian b/scripts/package/mkdebian
index 60a2a63a5e90..edab4503c105 100755
--- a/scripts/package/mkdebian
+++ b/scripts/package/mkdebian
@@ -139,13 +139,20 @@ echo $debarch > debian/arch
 extra_build_depends=", $(if_enabled_echo CONFIG_UNWINDER_ORC libelf-dev:native)"
 extra_build_depends="$extra_build_depends, $(if_enabled_echo CONFIG_SYSTEM_TRUSTED_KEYRING libssl-dev:native)"
 
+# Support SOURCE_DATE_EPOCH in changelog for reproducible builds
+packagetimestamp_opts=
+if [ -n "$SOURCE_DATE_EPOCH" ]; then
+    packagetimestamp_opts="-d @$SOURCE_DATE_EPOCH"
+fi
+packagetimestamp="$(date -R $packagetimestamp_opts)"
+
 # Generate a simple changelog template
 cat <<EOF > debian/changelog
 $sourcename ($packageversion) $distribution; urgency=low
 
   * Custom built Linux kernel.
 
- -- $maintainer  $(date -R)
+ -- $maintainer  $packagetimestamp
 EOF
 
 # Generate copyright file
